packages/rest/src/validation/request-query.validator.ts

```diff
@@ -0,0 +1,150 @@
+import {
+  RequestBodyObject,
+  SchemaObject,
+  ReferenceObject,
+  SchemasObject,
+} from '@loopback/openapi-v3-types';
+import * as AJV from 'ajv';
+import * as debugModule from 'debug';
+import * as util from 'util';
+import {HttpErrors, RestHttpErrors, RequestBody} from '..';
+import * as _ from 'lodash';
+
+const toJsonSchema = require('openapi-schema-to-json-schema');
+const debug = debugModule('loopback:rest:validation');
+
+export type RequestQueryValidationOptions = AJV.Options;
+
+/**
+ * Check whether the request query is valid according to the provided OpenAPI schema.
+ * The JSON schema is generated from the OpenAPI schema which is typically defined
+ * by `@requestQuery()`.
+ * The validation leverages AJS schema validator.
+ * @param query The request query parsed from an HTTP request.
+ * @param requestQuerySpec The OpenAPI requestQuery specification defined in `@requestQuery()`.
+ * @param globalSchemas The referenced schemas generated from `OpenAPISpec.components.schemas`.
+ */
+export function validateRequestQuery(
+  query: RequestBody,
```

Comment:
If we want to reuse `RequestBody` for query, we should rename the type as something like `ValueWithSchema`.

Reply:
@raymondfeng - do you mean we should duplicate the type requestBody in types.ts and then use this new type (ValueWithSchema) in request-query.validator.ts?  
should we also duplicate the BodyParser, BodyParserFunction? and then duplicate the code in body-parser.ts?

Reply:
No duplication. What I meant is to refactor `RequestBody` to `ValueWithSchema` into its own file and use it for both body and query validation.